Source/Models/APIClient.swift

```diff
@@ -0,0 +1,30 @@
+import Foundation
+import Argo
+import Result
+
+public struct APIClient {
+  private let requestPerformer: RequestPerformer
+
+  public init(requestPerformer: RequestPerformer = NetworkRequestPerformer()) {
+    self.requestPerformer = requestPerformer
+  }
+}
+
+public extension APIClient {
+  func performRequest<T: Request>(request: T, completionHandler: Result<T.ResponseType, NSError> -> Void) {
+    requestPerformer.performRequest(request.build()) { result in
+      let object = result >>- deserialize >>- { request.parse($0) }
+      completionHandler(object)
+    }
+  }
+}
+
+private func deserialize(response: HTTPResponse) -> Result<JSON, NSError> {
+  do {
+    let data = response.data ?? NSData()
```

Comment:
I _could_ do something like:

```swift
guard let data = response.data { return .Failure(.error("No response data")) }
```

That might be nicer than this, and would let us continue to use non-optionals inside the body of this function.